Puppi isolation electrons miniaod

ishvetso opened the pull request:
adding puppi isolation for electrons in miniAOD as userFloat(based and on puppi and puppiNoLeptons, calculated using puppiWeight() and puppiWeightNoLep() from packedCandidate). Added a possibility to configure puppi vs puppiNoLeptons in CITKPFIsolationSumProducerForPUPPI which should used in case the weights are taken from packedCandidate. In addition some renaming for puppi photon isolation in miniAOD(minor)

hroskes committed:
Use TkAlStyle in track splitting via the All In One tool
Shift the units of phi, theta, and q/pT

Marker style in track splitting

uhh2-bot committed:
Add common style for TkAl plots

hroskes committed:
Begin to use TkAlStyle for DMRs

Set axis styles by TkAlStyle

Put CMS title on DMRs, move detector id using code from Candice

Don't set canvas height manually, use TkAlStyle instead

hroskes committed:
Add configuration for a custom title to the All In One Tool

Rename UNSET to avoid ambiguity

Remove one time use segment

Use era in the All In One Tool

uhh2-bot committed:
Update common TkAl style and add test file with examples

hroskes committed:
CMS header in PubComm style.  In a custom title, #CMS{text} puts CMS in big bold and text in italics

hroskes committed:
Remove _{org}

y axis labels for histograms

Don't make run number/nHits plots, they're not that interesting

hroskes committed:
Legend header

hroskes committed:
Move plotting options to their own section

hroskes committed:
axis labels for DMRs

More information and 2 columns in DMR legend

Control over legend contents for DMRs

legendoptions for track splitting, and better calculation of the mean and rms

fixes for surface shape plots

Legend header for DMRs

jbsauvan committed:
Add test config to run on already produced HFCFETriggerDigi

jbsauvan committed:
Fix E -> ET conversion

jbsauvan committed:
Shorten data frame emplace_back

jbsauvan committed:
Use ForwardDetId instead of int

jbsauvan committed:
Update best choice unit test

jbsauvan committed:
Merge pull request #34 from jbsauvan/hgc-tpg-add-fhe

Add FH in TPG

jbsauvan committed:
Merged hgc-tpg-devel-CMSSW_8_1_0_pre3 from repository PFCal-dev

jbsauvan committed:
Update geometry config for pre6 and temporarily remove unused BH

jbsauvan committed:
Change auto_ptr to unique_ptr

jbsauvan committed:
Update member name in BE algos

jbsauvan committed:
Merged hgc-tpg-devel-CMSSW_8_1_0_pre6 from repository PFCal-dev

jbsauvan committed:
Remove data files for integration

jeongeun committed:
CleanUp wildcards on event content module

rovere committed:
Optimise detector grouping for material description

The detector grouping into same-material groups has been redesigned in
order to better follow the real material positioning, taking into
account that the material is equally shared among adjacent detectors.
All elements that are in direct contact either with support structures
or any other external source of material have been partitioned into
independent groups, so that the material is not uniformly shared on all
the layers-disks, but better localized.

rovere committed:
Protect if XML schema differs in release and local

rovere committed:
Improve graphics for grouping plots

rovere committed:
Add more samples to verify the RecoMaterial map

rovere committed:
Add more plots to monitor deltaP{,t,z}

jbsauvan committed:
Move stringstream outside loop and reserve vector

rovere committed:
WIP, to be tested

hroskes committed:
compareAlignments, stop earlier for phase 1

Ignore files with the wrong phase geometry when merging pixel

hroskes committed:
customrighttitle

alja committed:
Search for detail views for complete list of plugins instad of available plugins.

alja committed:
Allow more than 2 types of detail view.

alja committed:
Changes in legend.

alja committed:
New ECAL rechit detail view.

hroskes committed:
outlier cut as a cfg file option

Get mean and rms from histogram by default

hroskes committed:
Optional big text in the legend for offline validation

hroskes committed:
Fix legend for split DMRs

alja committed:
Separate filling for XY and EtaPhi histogram.

alja committed:
Add marker.

alja committed:
Handle numeric errors around -Pi , Pi.

alja committed:
Remove debug print.

alja committed:
set orthographic camera by the default.

alja committed:
draw EB and EE.

alja committed:
comment out marker

Commit:
Adding general track to ALCARECO for muonalign

Commit:
updating validation suite to have the possibility to monitor both OT and ITPixel digis

Commit:
add back ALCARECOStreamHcalCalIsolatedBunchFilter

alja committed:
Draw TEvBox for tower. Leave TEveCaloData empty.

alja committed:
add tooltip

alja committed:
remove oboslete code

alja committed:
debug superclusters.

alja committed:
remove EE specifics.

alja committed:
remove EE specifics. comment out addTrackPointsinCaloData.

alja committed:
set supercluster colors

alja committed:
Draw the tallest with a cross

alja committed:
Draw outline.

bartosik-hep committed:
Made JSON and CSV output optional

bartosik-hep committed:
Corrected usage info

alja committed:
Draw  3DCaloLego in conversion detail view.

alja committed:
Remove color palette shortcut.

alja committed:
Make track line width configurable.

alja committed:
Make supercluster colors a tone darker.

alja committed:
Put line and box in a compound.

alja committed:
Don't draw ecal barrel by the default.

alja committed:
have consistent box and line color.

alja committed:
Search for reducedEGamma collection : miniaod case.

alja committed:
Remove debug print

alja committed:
Skip consumers (dependent module registration) for eacal rec hit based detail views. In general need ecalRecHit_, reducedEcalRecHitsEB, or reducedEgamma. I presume the data will have one of those which is checked in FWECALDetailViewBuilder.

deguio committed:
remove old HCAL DQM

deguio committed:
clean central sequences

vkhristenko committed:
preping for a PR

Commit:
additional and modified plots for tracking DQM

Commit:
fixing logic

Commit:
retrying PR 15324

Commit:
Couple missing files from PR 15324

manoni committed:
restored INELASTIC_XSEC_13TeV value to 73.1

mtosi committed:
fix genericTriggerEventFlag

vkhristenko committed:
fixing phase1 crashes in the harvesting step and adding histograms for QIE10 monitoring requested by hcal performance study group

mtosi committed:
add dedicated plots for the HIP studies based on trigger selections

mtosi committed:
add monitoring VS LS

mtosi committed:
add MEs vs BX, lumi

Commit:
removed CosmicInMuon

makortel committed:
Use standard tracking validation for phase2

Reduces the set of histograms significantly. The additional histograms
of trackingOnlyValidation are interesting when tuning tracking, but
are not necessary for release validation.

vciulli committed:
Fix setting cross section in Rivet analaysis

bartosik-hep committed:
BugFix: progress counter crashing with <10 events

hroskes committed:
chi2 and prob plots are better

statbox fixed

Commit:
remove pathALCARECOMuAlGlobalCosmicsInCollisionsTracks

vkhristenko committed:
removing using namespaces from headers of non-hcaldqm

Commit:
bTag validation for PhaseII reco

hroskes committed:
surface shape style

hroskes committed:
Get the package from the release if no addpkg

Various fixes

rovere committed:
Improve Reco Material tools

rovere committed:
Add inner and outer information to Track Ntuple

olivito committed:
adding dqm for double tkmu and aux paths

matz-e committed:
Change HCAL TP compression to account for multiple FG bits.

HCAL TP in 2017 have up to 4 fine-grain bits.  The
HcalTriggerPrimitiveSample already supports setting multiple fine-grain
bits, this adds support to the various compression classes.

In preparation for the 4 HE feature bits of 2017.

fioriNTU committed:
TkMapFix

fioriNTU committed:
modified according to Gaelle comment

Commit:
Rebased on #15466

Dr15Jones committed:
Do not signal a delayed get when doing prefetching from the Source

Removed the signal for delayed get from EventPrincipal::readFromSource_ and instead issue it from the InputProductResolver. The InputProductResolver can tell the difference between a prefetch and an actual delayed get. This change also allowed the signals to be removed entirely from EventPrincipal since they were already available for unscheduled execution handling.

Dr15Jones committed:
SubProcess does not need read from source signal

The SubProcess can not read from the source directly (it is done indirectly through the EventProcessor) and therefore its EVentPrincipal does not need that signal set.

Dr15Jones committed:
Moved signals into InputSource

Moved the preEventReadFromSourceSignal and postEventReadFromSourceSignal into the InputSource. The InputSource is now responsible for passing those signals to the DelayedReader. Previously these signals were in EventPrincipal.

Commit:
Add DB objects for HCAL TP

Commit:
Resolve merge attempt

Dr15Jones committed:
Removed readFromSource from Principal

The InputProductResolver directly calls the DelayedReader to do its work. Previously only the EventPrincipal actually had a working readFromSource. This was preserved in order to avoid a crash if the file has been closed before the delayed read was attempted.

Commit:
Recover merging process

Commit:
Some more missed out changes

yetkinyilmaz committed:
boost bugfix

rovere committed:
New Tracker Material, also accounting for external material

rovere committed:
WIP

rovere committed:
Import PhaseI XML from 81X branch

rovere committed:
First prototype of RecoMaterial Tool

A first prototype of a RecoMaterial tool is introduced. It
has also been used to derive a better grouping of detectors
for the Run2 geometry. The changes to the XML have not been
propagated centrally but only kept as a reference in the
SimTracker/TrackerMaterialAnalysis package.

rovere committed:
Improve help message

Dr15Jones committed:
Protected against possible null signal

It is possible that the specific DelayedReader does not have the signals set and therefore we should not crash.

Commit:
Fix issue for LUT

ishvetso committed:
add CITK producers for electrons

ishvetso committed:
add module for electrons

ishvetso committed:
fix name

ishvetso committed:
update for PUPPI isolation

Commit:
After cure from Chris

Dr15Jones committed:
Principal no longer directly gets data from DelayedReader

The DelayedReader is only used to get data via the InputProductResolver. Previously merging was handled by the Principal directly which was not properly setting the status in the InputProductResolver in the case where the data was not obtained.

Dr15Jones committed:
Moved locking shared resources out of DelayedReader and into InputProductResolver

By moving the lock to InputProductResolver we are able to check to see if while we were waiting for the lock another thread already retrieved the same data product so we do not have to call DelayedReader::getProduct again. This also makes the InputProductResolver::putProduct call safe from multiple threads since it is only called while in the lock.

Dr15Jones committed:
Moved make_sentry to own header file

ishvetso committed:
Merged PUPPI_isolation_electrons_miniAOD from repository ishvetso

ishvetso committed:
fix naming

ishvetso committed:
update for puppi isolation for electrons+renaming

cmsbuild commented:
A new Pull Request was created by  @ishvetso (Ivan Shvetsov) for CMSSW_8_1_X.

It involves the following packages:

PhysicsTools/IsolationAlgos
PhysicsTools/PatAlgos
RecoEgamma/EgammaIsolationAlgos


@cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please review it and eventually sign? Thanks.
@rappoccio, @Sam-Harper, @rafaellopesdesa, @imarches, @ahinzmann, @acaudron, @mmarionncern, @TaiSakuma, @lgray, @jdolen, @nhanvtran, @gpetruc, @schoef, @ferencek, @mverzett, @mariadalfonso, @pvmulder, @JyothsnaKomaragiri this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

Dr15Jones committed:
Protect against possible null value for aux_

The MixingModule does not setup the unscheduled system and therefore InputProductResolver::aux_ does not get set.

VinInn committed:
abort if TSFF is full

VinInn committed:
fix ABrt counter

VinInn committed:
add logError

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: c6a0bafd6ead8f8a96c0af2a828f0caac4ba1254
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15528/14614/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15528/14614/summary.html

cmsbuild committed:
Merge pull request #15526 from Dr15Jones/haveInputProductResolverUseDelayedReader

Have InputProductResolver use DelayedReader

cmsbuild committed:
Merge pull request #15462 from vciulli/fixRivetCrossSection

Fix setting cross section in Rivet analaysis

davidlange6 committed:
Merge pull request #15514 from matz-e/hcal-tp-compression

Change HCAL TP compression to account for multiple FG bits.

cmsbuild committed:
Merge pull request #15447 from mtosi/hipMonitoring

hip monitoring

davidlange6 committed:
Merge pull request #15448 from makortel/trkStandardValidationPhase2

Use standard tracking validation for phase2

cmsbuild committed:
Merge pull request #15227 from suchandradutta/Phase2Digitizer_81X_160718

Digi validation suite updated

mtosi committed:
fix normalization for empty plots

davidlange6 committed:
Merge pull request #14831 from PFCal-dev/hgc-tpg-integration-81X

Updates in HGCAL L1 trigger

Martin-Grunewald committed:
TSG test files from TSG eos location (81X)

mmusich committed:
Miscellaneous fixes to new FSQ vtx smearing parameters

monttj commented:
@ishvetso 

Dear Ivan,

I am wondering if adding more variables should be in upper level PAT object here.
https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/PhysicsTools/PatAlgos/plugins/PATPhotonProducer.cc

I might miss something. But the photon slimmer is for slimming PAT object not adding more variables?

Regards,
Taejeong

ishvetso commented:
Dear Taejenog,
well, I believe one can add new variables to the pat object, while the intension of this PR was to add a new variable on the miniAOD level.

Regards,
Ivan.

rovere committed:
Revert Labels-Component logic

Instead of specifying a component by name, the user has to
specify it by label. This makes the interplay with others
scripts much easiers, since links are mostly guaranteed to
be by label and not by component. Also, combined components
are somewhat easier to handle.

rovere committed:
Reformatting

Commit:
Increase pt threshold of L2Mu NoBPTX path

surnairit committed:
added HLT_DoubleMu0

rovere committed:
Add Tools to compare MC and Reco material

A bunch of scripts and ROOT macros have been added and properly
documented to guide the user in the process of "validating" the
reconstrucion material as input to the reconstruction step via either
the DB or via local XML files.

rovere committed:
Restore current Material for Run2 Detector

rovere committed:
Revert to PhaseI geometry as default

rovere committed:
Revert to PhaseI geometry as default

rovere committed:
Remove private script

rovere committed:
Force Markdown rendering by extension

rovere committed:
Add more verbose instructions

rovere committed:
Improve comments and documentation

rovere committed:
Remove wrongly added file

rovere committed:
Initialize variables

Commit:
Second rebase

hroskes committed:
Layers for phase 0 or 1

kurtejung committed:
removed final cout

smuzaffar committed:
make use of internal cmssw_das_client

smuzaffar committed:
added cmssw_das_client.py whcih calls das_client wrapper

mbandrews committed:
Include kUnknown integrity in Global Qt

mbandrews committed:
Show BAD in RawData when FE is DISABLED

hroskes committed:
Style Z->mumu

Specify marker style for Z->mumu like for track splitting

slava77 commented:
I tend to agree with @monttj 
It is  more appropriate to add these variables at "reco" step of PAT.
Slimming is meant to set already available values to "zeros" or other "packed" values.

Unless there is some circular logic involved to add the userFloats in PATElectronProducer (as pat electrons are needed to compute the iso), I think we should move the electron and photon puppi isolation value settings to corresponding PAT{Electron,Photon}Producer

alja committed:
Remove obsolete code.

Commit:
Changes in TB configuration and removal of hardwired layer grouping

hroskes committed:
testValidate

ishvetso commented:
ok, I see. One question to that: this should be done in the same manner? I mean like  
electron.addUserFloat("PUPPI_Isolation", PUPPI_Isolation);

slava77 commented:
On 8/22/16 12:37 PM, Ivan Shvetsov wrote:
> ok, I see. One question to that: this should be done in the same manner?
> I mean like
> 
> electron.addUserFloat("PUPPI_Isolation", PUPPI_Isolation);

Yes, it should work the same.

The alternative is to add it as user isolation,
but it may be less trivial to modify.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-241525149>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbkLkoPBNdd6L44V_Zf4Jujz5Piaqks5qifqCgaJpZM4JpENC>.
>

hroskes committed:
gitignore for dataset files automatically created by the all in one tool

afiqaize committed:
Added the customizeHLTforCMSSW 81X

smuzaffar committed:
remove unused variable to avoid gcc493 build errors

mmusich committed:
Adding key for Run2 pA simulation

ishvetso commented:
I have just realized one thing. One of reasons to put this in PATElectron(Photons)Slimmer.cc was the access to puppiWeight directly from packedCandidate. This is useful from the timing point of view (I think we discussed in some time before). Otherwise, we need to get weight from puppi value map. I'm not sure if this is included.

gpetruc commented:
Hi Ivan,

The packed candidates don't depend on the pat objects, only on pf
candidates and slimmed vertices, so it should be possible to use them
before the production of the pat leptons (and the framework should take
care of the schedule)

Giovanni

Il 23-ago-2016 10:09, "Ivan Shvetsov" <notifications@github.com> ha scritto:

> I have just realized one thing. One of reasons to put this in
> PATElectron(Photons)Slimmer.cc was the access to puppiWeight directly from
> packedCandidate. This is useful from the timing point of view (I think we
> discussed in some time before). Otherwise, we need to get weight from puppi
> value map. I'm not sure if this is included.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-241658391>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEbbR0XcXbaV9GulECpDO1MneigKkpDwks5qiqrSgaJpZM4JpENC>
> .
>

ishvetso commented:
Hi Giovanni,
 
I see. And the userFloat should be written to patElectrons or selectedPatElectrons?

Regards,
Ivan.

civanch committed:
added more debug capabilities

violatingcp committed:
updated 81X with fix from PR 15429. Aim is to fix some issues with the photon performance and Puppi

cmsbuild committed:
Merge pull request #15559 from smuzaffar/fix-gcc493-unused-variable-error

remove unused variable to avoid gcc493 build errors

cmsbuild committed:
Merge pull request #15553 from mbandrews/GlobalQt81X

Set Global Quality to show RED when FE is disabled

cmsbuild committed:
Merge pull request #15548 from kurtejung/boostdebug_81X_04

Fix boosting software for pPb MC

cmsbuild committed:
Merge pull request #15494 from cms-btv-pog/Phase2bTagValidation_from-CMSSW_8_1_0_pre9

bTag validation for PhaseII reco

davidlange6 committed:
Merge pull request #15549 from smuzaffar/cmssw_das_client

Internal das client python module to call das_client wrapper

cmsbuild committed:
Merge pull request #15546 from surnairit/DoubleMu0_DQM_81X

Modifications to include HLT_DoubleMu0 path in the HeavyFlavor DQM package in CMSSW_8_1_X

davidlange6 committed:
Merge pull request #15529 from VinInn/FixTSFF

Fix throwing CapacityExausted

cmsbuild committed:
Merge pull request #15539 from Martin-Grunewald/TSGeos81X

TSG test files from TSG eos location (81X)

cmsbuild committed:
Merge pull request #15541 from mmusich/81X_FSQ_vtxSmearingFixes

Miscellaneous fixes to new FSQ vtx smearing parameters

cmsbuild committed:
Merge pull request #15555 from bsunanda/Run2-TB06

bsunanda:Run2-TB06 Changes in TB configuration and removal of hardwired layer grouping

cmsbuild committed:
Merge pull request #15560 from mmusich/81X_addKeyForpPb

Adding key for Run2 pA simulation

gpetruc commented:
patElectrons

In fact, I think it could make sense to add the isolation as data member,
not userFloat. What do others think?
(tagging also @arizzi in case he's not following already)

Giovanni

On Tue, Aug 23, 2016 at 12:00 PM, Ivan Shvetsov <notifications@github.com>
wrote:

> Hi Giovanni,
>
> I see. And the userFloat should be written to patElectrons or
> selectedPatElectrons?
>
> Regards,
> Ivan.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-241684564>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEbbRwv6952sNcVlrGH7ZkLRoLJJSUTdks5qisS-gaJpZM4JpENC>
> .
>

ishvetso commented:
as data member means adding data new data member to:
DataFormats/PatCandidates/interface/Lepton.h

and also adding functions to set the corresponding values (as far as I see)?

gpetruc commented:
Yes


On Tue, Aug 23, 2016 at 2:42 PM, Ivan Shvetsov <notifications@github.com>
wrote:

> as data member means adding data new data member to:
> DataFormats/PatCandidates/interface/Lepton.h
>
> and also adding functions to set the corresponding values (as far as I
> see)?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-241717762>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEbbRxXLOmuCVs0TKfB8Ep4qZq3DsSsfks5qiuqngaJpZM4JpENC>
> .
>

ishvetso commented:
for electrons there is puppi + puppiNoLeptons, for photons we studies only puppi, so we don't have puppiNoPhoton for the photon case.
So do we want to have puppiNoLeptons for photon? Meaning that since puppi and puppiNoLepton weights are available in packedCandidate. So it's cheap to have it, though only one can be used for photons etc. Are we ok with this?

cmsbuild committed:
Merge pull request #15221 from lpernie/AddgeneralTrackALCARECOmuon

Adding general track to ALCARECO for muonalign

cmsbuild committed:
Merge pull request #15380 from vkhristenko/hcaldqm_phase1_updates

HCAL DQM Updates

afiqaize committed:
use std::abs, cache the eta 81X

bartosik-hep committed:
Output files are created even if no affected events found, if required by flags

ishvetso committed:
Merged PUPPI_isolation_electrons_miniAOD from repository ishvetso

bartosik-hep committed:
Recovered summary print out, but only with affected paths. Can be disabled with -q option

hroskes committed:
print number of modules outside range

cmsbuild committed:
Merge pull request #15511 from olivito/dqm_doubletkmu_81x

HLT DQM for double tkmu and aux paths, 81X

cmsbuild committed:
Merge pull request #15545 from jruizvar/L2Mu81X

Increase pt threshold of L2Mu NoBPTX path (81X)

cmsbuild committed:
Merge pull request #15374 from deguio/cleanHcalDQM

Clean hcal dqm

davidlange6 committed:
Merge pull request #15531 from mtosi/fixV0Monitoring

V0 monitoring: fix normalization for empty plots

hroskes committed:
further fixes

Dr15Jones committed:
Added BusyWaitIntLegacyProducer for tests

The module was added to allow testing of stalling when running multiple threads.

Dr15Jones committed:
Update edmStreamStallGrapher

The update deals with the framework change which changed when signals occur as well as allowing multiple modules to run on each stream.

Commit:
Make use of HcalFrontEndMap

Commit:
Checking neutrino tracking and saving info

rovere committed:
Use LogInfo instead of plain std::cout

rovere committed:
Fix static analyzer warnings

rovere committed:
Comment apparently unused functions

Dr15Jones committed:
Wrote unit test for edmStreamStallGrapher.py

ishvetso commented:
may be one more thing that I would like to understand. That should be implemented as userIsolation() ? Adding new key in DataFormats/PatCandidates/interface/Isolation.h 

just to clarify otherwise it seems to lead to a lot of changes like  in
PhysicsTools/PatAlgos/interface/MultiIsolator.h

violatingcp committed:
reverted back to 81X format

monttj commented:
Hello,

userIsolation is for users in case they want to have customized isolation values. 
I would let analyzers use this space. 

Given that the default particle flow isolation is still the one we use in analysis, I think you can create another function like "puppiChargedHadronIso()", "puppiNeutralHadronIso()" and "puppiPhotonIso()". But I am not sure if the puppi isolation can be implemented in the same way it is done for default particle flow isolation in PAT. Because the default isolation was done using isodeposit while new implementation from egamma is based on CITK. 

By the way, photon object is not from Lepton.h class.

Taejeong

ianna committed:
Rebasing beampipe changes

cmsbuild committed:
Merge pull request #15574 from Dr15Jones/updateEdmStreamStallGrapher

Update edmStreamStallGrapher.py

cmsbuild committed:
Merge pull request #15561 from civanch/extended_g4_debug_capabilities

Added more SIM debug capabilities

cmsbuild committed:
Merge pull request #15517 from fioriNTU/TkMapFix81X

Fix for Tracker Maps and Shifter Summary 81X

cmsbuild committed:
Merge pull request #15521 from bsunanda/Run2-hcx89

bsunanda:Run2-hcx89 Add DB objects for HCAL TP

cmsbuild committed:
Merge pull request #15440 from afiqaize/rhoCorr81X

rhoCorr binned in eta 81X, apply rhoCorr for H/E

cmsbuild committed:
Merge pull request #15397 from manoni/CMSSW_8_1_X_2016-08-08-2300-BX

additional and modified plots for SiStrip and Tracking DQM

ishvetso commented:
ok, so are we ok to introduce to 6 functions: charged hadrons, neutral, photons, for puppi and puppiNoLeptons? I don't know if it is ok to have as many functions for settings. And as mentioned by Taejenog photon and electron are different objects so that has to be doubled ... I don't know if we can avoid it

ishvetso commented:
also tagging @fcouderc and @ikrav

davidlange6 committed:
Merge pull request #15527 from alja/ECAL-DV-81-merge

81x Fireworks: new version of ECAL detail view

ishvetso committed:
commit for review

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso commented:
I have did a commit which is more or less my understanding for the moment, can you please check that?
I actually got an error : 
recipe for target 'tmp/slc6_amd64_gcc530/src/DataFormats/PatCandidates/src/DataFormatsPatCandidates/libDataFormatsPatCandidates.so' failed

slava77 commented on PhysicsTools/IsolationAlgos/plugins/CITKPFIsolationSumProducerForPUPPI.cc:
```diff
@@ -74,6 +75,7 @@ namespace citk {
         useValueMapForPUPPI = true;
       }
       else useValueMapForPUPPI = false;
+      if (c.exists("usePUPPINoLepton")) usePUPPINoLepton = c.getParameter<bool>("usePUPPINoLepton");
```
::exists method is rather an exception left for cases for rather old and complex pre-7X software.
Define a ::fillDescriptions method https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp

ishvetso replied:
should be fixed now

slava77 commented on PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc:
```diff
@@ -771,7 +785,9 @@ void PATElectronProducer::fillElectron(Electron& anElectron,
 
+
   for (size_t j = 0; j<isolationValues.size(); ++j) {
+    std::cout << " test iso: " << isolationValueLabels_[j].first << std::endl;
```
this will have to be removed

ishvetso replied:
fixed

slava77 commented on DataFormats/PatCandidates/interface/Electron.h:
```diff
@@ -160,6 +160,17 @@ namespace pat {
 
+      float puppiChargedHardons() const {return puppiChargedHardons_; };
+      float puppiNeutralHardons() const {return puppiNeutralHardons_; };
+      float puppiPhotons() const {return puppiPhotons_; };
```
these have to say that it is isolation

ishvetso replied:
fixed

ishvetso committed:
rename

slava77 commented on PhysicsTools/PatAlgos/plugins/PATElectronSlimmer.cc:
```diff
@@ -114,10 +127,34 @@ pat::PATElectronSlimmer::produce(edm::Event & iEvent, const edm::EventSetup & iS
+        double PUPPI_Isolation = (*PUPPIIsolation_charged_hadrons)[elePtr] + (*PUPPIIsolation_neutral_hadrons)[elePtr] + (*PUPPIIsolation_photons)[elePtr];
+        double PUPPINoLeptons_Isolation = (*PUPPINoLeptonsIsolation_charged_hadrons)[elePtr] + (*PUPPINoLeptonsIsolation_neutral_hadrons)[elePtr] + (*PUPPINoLeptonsIsolation_photons)[elePtr];
+        electron.addUserFloat("PUPPI_Isolation", PUPPI_Isolation);
+        electron.addUserFloat("PUPPINoLeptons_Isolation", PUPPINoLeptons_Isolation);
```
the idea for recent updates was that all changes will be removed from the ElectronSlimmer

ishvetso committed:
remove cout

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso commented:
There is one potential problem: for the photons we should arrive to this line
https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/RecoEgamma/EgammaIsolationAlgos/plugins/isolation_cones/PhotonPFIsolationWithMapBasedVeto.cc#L135

which means we need associatedPackedPFCandidates()
and PATPhotonProducer has gedPhotons as input

but I hope conversion to pat::Photon is going to work:
 pat::patPhotonPtr aspat_photonptr(photon)
?

slava77 commented:
On 8/24/16 4:47 AM, Ivan Shvetsov wrote:
> There is one potential problem: for the photons we should arrive to this
> line
> https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/RecoEgamma/EgammaIsolationAlgos/plugins/isolation_cones/PhotonPFIsolationWithMapBasedVeto.cc#L135
> 
> which means we need associatedPackedPFCandidates()
> and PATPhotonProducer has gedPhotons as input
> 
> but I hope conversion to pat::Photon is going to work:
> pat::patPhotonPtr aspat_photonptr(photon)
> ?

As I said earlier,
the move of isolation setting to the layer1 producer will work only if
there is no circular logic.
An example of circular logic is  the making of isolation needs objects
from layer1 producer or later.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-242036084>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbgkRj6tScHCoNMdDt-mGy_6axd-Rks5qjC9JgaJpZM4JpENC>.
>

slava77 commented:
I'm catching up with recent discussion/development now.
It looks like we have inadvertently extended the scope of the original PR.

There are two features now
- addition of the puppi isolation floats to the standard interface of pat::Electron and pat::Photon (why would it not eventually apply for Muon?)
- filling of this variable at the layer-1 pat producer step

These work well for running on AOD.

Do we need support for running on top of miniAOD as well?

hroskes committed:
Remove commented section

ishvetso commented:
24.08.16 14:00, Slava Krutelyov пишет:
> On 8/24/16 4:47 AM, Ivan Shvetsov wrote:
> > There is one potential problem: for the photons we should arrive to this
> > line
> > 
> https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/RecoEgamma/EgammaIsolationAlgos/plugins/isolation_cones/PhotonPFIsolationWithMapBasedVeto.cc#L135
> >
> > which means we need associatedPackedPFCandidates()
> > and PATPhotonProducer has gedPhotons as input
> >
> > but I hope conversion to pat::Photon is going to work:
> > pat::patPhotonPtr aspat_photonptr(photon)
> > ?
>
> As I said earlier,
> the move of isolation setting to the layer1 producer will work only if
> there is no circular logic.
> An example of circular logic is the making of isolation needs objects
> from layer1 producer or later.
But then I assume it is not going to work?Right? 
associatedPackedPFCandidates() should be written at layer1 but I'm not 
100% sure
if this is the case, should we consider the initial idea of userFloat()?
>
>
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-242036084>, or
> > mute the thread
> > 
> <https://github.com/notifications/unsubscribe-auth/AEdcbgkRj6tScHCoNMdDt-mGy_6axd-Rks5qjC9JgaJpZM4JpENC>.
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-242038764>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFMRQnJAFrUulQIhjc4hVlQ7Y5cdRag4ks5qjDJ1gaJpZM4JpENC>.
>

-- 
Best regards,
Ivan Shvetsov.

ishvetso commented:
concerning this comment:
>There are two features now
addition of the puppi isolation floats to the standard interface of pat::Electron and pat::Photon (why would it not eventually apply for Muon?)
filling of this variable at the layer-1 pat producer step
These work well for running on AOD.

- what is not clear to me is how this is going to work on AOD since we need packedPFCandidate
- there is a possibility not to depend on packedPFCandidates but then we need puppi, and in this case we avoid problem(?) with associatedPackedPFCandidates() that I mentioned above
- in case we run puppi I think there would be a problem with puppiNoLeptons which I didn't manage to solve, that's relevant for electrons only. May consider having only puppi isolation
- for muons study was done by Satoshi and I did study for electrons and photons. But I think we should first understand how/what exactly we do.

slava77 commented:
On 8/24/16 5:37 AM, Ivan Shvetsov wrote:
> concerning this comment:
> 
>     There are two features now
>     addition of the puppi isolation floats to the standard interface of
>     pat::Electron and pat::Photon (why would it not eventually apply for
>     Muon?)
>     filling of this variable at the layer-1 pat producer step
>     These work well for running on AOD.
> 
>   * what is not clear to me is how this is going to work on AOD since we
>     need packedPFCandidate

My comment referred to rerunning miniAOD on top of AOD.
This is how code from this PR works now.

For running on miniAOD inputs,
a more minimal solution using CITK is more appropriate.
(compute iso using saved miniAOD objects -> modify objects to add
isolation variables).


>   * there is a possibility not to depend on packedPFCandidates but then
>     we need puppi, and in this case we avoid problem(?) with
>     associatedPackedPFCandidates() that I mentioned above
>   * in case we run puppi I think there would be a problem with
>     puppiNoLeptons which I didn't manage to solve, that's relevant for
>     electrons only. May consider having only puppi isolation
>   * for muons study was done by Satoshi and I did study for electrons
>     and photons. But I think we should first understand how/what exactly
>     we do.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-242047341>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbj0gDwgpIZ20rjD8tNdl8gnBRi8rks5qjDsAgaJpZM4JpENC>.
>

bartosik-hep committed:
Switched to C++ libraries include. Simple printf -> cout

Commit:
GT for pA 2016

Dr15Jones committed:
Handle case where message order is inverted from Tracer

Under heavy output to the message logger, the order of some messages
can become inverted. That is a module 'finished' message can be printed
before the corresponding module 'started'. The code now catches this
case and ignores those messages.

civanch committed:
fixed static analyzer warnings

cmsbuild committed:
Merge pull request #15577 from ianna/post-ls2-beam-pipe

Rebasing Post LS2 Beampipe

davidlange6 committed:
Merge pull request #14853 from jeongeun/FixPR14406

CleanUp wildcards on event content module(Fix PR14406)

hroskes committed:
Merge branch 'TkAlPublicPlots-80X-PR' of https://github.com/hroskes/cmssw into TkAlPublicPlots-81X-PR

Conflicts:
	Alignment/OfflineValidation/python/TkAlAllInOneTool/betterConfigParser.py
	Alignment/OfflineValidation/python/TkAlAllInOneTool/configTemplates.py
	Alignment/OfflineValidation/python/TkAlAllInOneTool/offlineValidationTemplates.py
	MuonAnalysis/MomentumScaleCalibration/test/Macros/RooFit/MultiHistoOverlapAll_Z.C
	MuonAnalysis/MomentumScaleCalibration/test/Macros/RooFit/MultiHistoOverlap_Z.C

ishvetso committed:
minor

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso commented:
can you also comment on this error I get:

error: class 'pat::Electron' has a different checksum for ClassVersion 33. Increment ClassVersion to 34 and assign it to checksum 3659401979
Suggestion: You can run 'scram build updateclassversion' to generate src/DataFormats/PatCandidates/src/classes_def_objects.xml.generated with updated ClassVersion
config/SCRAM/GMake/Makefile.rules:1937: recipe for target 'tmp/slc6_amd64_gcc530/src/DataFormats/PatCandidates/src/DataFormatsPatCandidates/libDataFormatsPatCandidates.so' failed
gmake: *** [tmp/slc6_amd64_gcc530/src/DataFormats/PatCandidates/src/DataFormatsPatCandidates/libDataFormatsPatCandidates.so] Error 1

slava77 commented:
On 8/24/16 3:00 PM, Ivan Shvetsov wrote:
> can you also comment on this error I get:
> 
> error: class 'pat::Electron' has a different checksum for ClassVersion
> 33. Increment ClassVersion to 34 and assign it to checksum 3659401979


I suggest to follow these instructions.


> Suggestion: You can run 'scram build updateclassversion' to generate
> src/DataFormats/PatCandidates/src/classes_def_objects.xml.generated with
> updated ClassVersion
> config/SCRAM/GMake/Makefile.rules:1937: recipe for target
> 'tmp/slc6_amd64_gcc530/src/DataFormats/PatCandidates/src/DataFormatsPatCandidates/libDataFormatsPatCandidates.so'
> failed
> gmake: ***
> [tmp/slc6_amd64_gcc530/src/DataFormats/PatCandidates/src/DataFormatsPatCandidates/libDataFormatsPatCandidates.so]
> Error 1
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-242222059>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbkSIvJQD1LSmcwfR8j3-yGNKp-IAks5qjL8UgaJpZM4JpENC>.
>

ishvetso committed:
few fixes

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso commented:
ok, few fixed are done, but I get:
Illegal parameters found in configuration.  The parameters are named:
 'puppiIsolationChargedHadrons'
 'puppiIsolationNeutralHadrons'
 'puppiIsolationPhotons'
You could be trying to use parameter names that are not
allowed for this plugin or they could be misspelled.

when running cmsRun patTuple_onlyElectrons_cfg.py 

may be missed smth simple ...

slava77 commented:
On 8/24/16 3:51 PM, Ivan Shvetsov wrote:
> ok, few fixed are done, but I get:
> Illegal parameters found in configuration. The parameters are named:
> 'puppiIsolationChargedHadrons'
> 'puppiIsolationNeutralHadrons'
> 'puppiIsolationPhotons'
> You could be trying to use parameter names that are not
> allowed for this plugin or they could be misspelled.

It is unclear from this message which module is complaining about the
parameters.
I suspect it's the PATElectronProducer.
The fillDescriptions method should know about the parameters

> 
> when running cmsRun patTuple_onlyElectrons_cfg.py
> 
> may be missed smth simple ...
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-242233128>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbjghg1C5J7hVsU9g40d9wHb_Hvhjks5qjMsHgaJpZM4JpENC>.
>

monttj commented on PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc:
```diff
@@ -323,6 +336,7 @@ void PATElectronProducer::produce(edm::Event & iEvent, const edm::EventSetup & i
 	  const edm::RefToBase<reco::GsfElectron>& elecsRef = electrons->refAt(idx);
 	  Electron anElectron(elecsRef);
 	  anElectron.setPFCandidateRef( pfRef  );
+    anElectron.setIsolationPUPPI((*PUPPIIsolation_charged_hadrons)[elePtr], (*PUPPIIsolation_neutral_hadrons)[elePtr], (*PUPPIIsolation_photons)[elePtr]);
```
The same implementation needs to be done for not "useParticleFlow_". 
Under this condition, only electrons identified as electron in Particle flow algorithm will have the puppi isolation. 
But we would like to have the puppi isolation for all gedElectrons.

ishvetso replied:
are you sure about that? I thought it has to do with whether candidate is matched to the PF Candidate, so I would expect that we need it somewhere here:
https://github.com/ishvetso/cmssw/blob/1e180758f9416a797f0726c96769674d8cac1574/PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc#L505

monttj replied:
The track matching is needed to identify the PF candidate as the electrons in PF are track seeded electrons. And I believe that the option was there for full particle flow object based analysis including electrons and muons. 

The collection of gedGsfElectrons also contains ecal seeded electrons.  Some of these electrons would survive our usual offline cut. So it does not need to be matched to PF candidate unless one really wants to do the analysis with full particle flow objects. Besides, the default value for useParticleFlow is "False".

ishvetso replied:
I think now logic should be fine.

ishvetso committed:
fix fillDescription

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso commented:
fixing fillDescription solved the problem partially, now I got 
A std::bad_alloc exception was thrown.
I guess it caused by the fact that isolation sums are not in the sequence. How exactly this has to be done for AOD?

davidlange6 committed:
Merge pull request #15593 from civanch/fix_static_anal_warn7

Fixed static analyzer warnings7

cmsbuild committed:
Merge pull request #15591 from Dr15Jones/stallGrapherHandleMessageInversion

Handle case where message order is inverted from Tracer

cmsbuild committed:
Merge pull request #15589 from franzoni/gf-from-CMSSW_8_1_X_2016-08-24-1100

GT for pA 2016

cmsbuild committed:
Merge pull request #15588 from bartosik-hep/CMSSW_8_1_X_hltDiff_json

Refined hltDiff output for automatisation [8_1_X]

davidlange6 committed:
Merge pull request #15419 from igv4321/realistic-phase1-hf-reco-retry

Realistic phase1 hf reco retry

cmsbuild committed:
Merge pull request #15572 from bsunanda/Run2-sim01

bsunanda:Run2-sim01 Checking neutrino tracking and saving info

cmsbuild committed:
Merge pull request #15569 from bsunanda/Run2-hcx90

bsunanda:Run2-hcx90 Make use of HcalFrontEndMap

ishvetso committed:
logic fixed

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso committed:
fill description is added

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso committed:
PUPPI iso disregarded

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso commented:
>For running on miniAOD inputs,
a more minimal solution using CITK is more appropriate.
(compute iso using saved miniAOD objects -> modify objects to add
isolation variables).

concerning this point: I think I still don't have the complete picture. The way the current PR is working is taking the AOD as input? So this means that changes will appear only in miniAOD?

One more thing: as I mentioned above for photons input from pat photon is needed, do you say that for photons we are ok with userFloat solution? Meaning what was done in the PR ##15375

cmsbuild committed:
Merge pull request #15547 from rovere/RecoMaterialTools

Reco material tools

slava77 commented:
On 8/25/16 3:45 AM, Ivan Shvetsov wrote:
>     For running on miniAOD inputs,
>     a more minimal solution using CITK is more appropriate.
>     (compute iso using saved miniAOD objects -> modify objects to add
>     isolation variables).
> 
> concerning this point: I think I still don't have the complete picture.
> The way the current PR is working is taking the AOD as input? So this
> means that changes will appear only in miniAOD?
> 
> One more thing: as I mentioned above for photons input from pat photon
> is needed, do you say that for photons we are ok with userFloat
> solution? Meaning what was done in the PR ##15375
> <https://github.com/cms-sw/cmssw/pull/15375>

Which input is needed from pat for photons?


I think the same setup should apply to photons as for electrons.

> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-242346667>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbrDDvI9-F8AiAU1baKQhsNZLzdKMks5qjXIwgaJpZM4JpENC>.
>

ishvetso commented:
>Which input is needed from pat for photons?
associatedPackedPFCandidates() is needed

>I think the same setup should apply to photons as for electrons.
I agree, the point is that different footprint removal is done.

davidlange6 committed:
Merge pull request #15562 from violatingcp/puppiphotonmetfix81X

Puppi Photon update see PR 15429

slava77 commented:
On 8/25/16 7:15 AM, Ivan Shvetsov wrote:
>     Which input is needed from pat for photons?
>     associatedPackedPFCandidates() is needed

I see.
If I'm not mistaken, this is not an integral part of pat photon reco,
something available here for convenience.
It should be possible to get the associated packed cands for any gedPhotons


> 
>     I think the same setup should apply to photons as for electrons.
>     I agree, the point is that different footprint removal is done.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-242402130>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbmKiAHOSpZZ2q2197udshgB9X43yks5qjaN5gaJpZM4JpENC>.
>

davidlange6 committed:
Merge pull request #15570 from hroskes/TkAlPublicPlots-81X-PR

Alignment plotting style and other fixes

ishvetso commented:
ok, good so should work for photons as well.

I have another technical question: I need add the isolation sums in the sequence and I'm not sure how exactly this should be loaded? 
I'm running patTuple_standard_cfg.py and it also doesn't have packedPFCandidates in the schedule and this is needed for isolation sums

ishvetso committed:
add puppi isolation

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso committed:
line was removed accidentially

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso committed:
add puppi isolation to PATPhotonProducer

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso committed:
fix vint32

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso committed:
fix input

ishvetso committed:
fix patElectrons and patPhotons

cmsbuild commented:
Pull request #15528 was updated. @cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please check and sign again.

ishvetso commented:
there are some problems with testing the current PR.
first, just running the current PR gives segmentation violation which is related to associatedPackedCanidates(), 
then, commenting out the usage of associatedPackedCanidates() gives the problem from PATElectronProducer with the following message:

......
efetching for unscheduled module ShiftedParticleProducer/'shiftedPatUnclusteredEnUp'
   [142] Prefetching for unscheduled module CandPtrProjector/'pfCandsForUnclusteredUnc'
   [143] Prefetching for unscheduled module CandPtrProjector/'pfCandsNoJetsNoEleNoMuNoTau'
   [144] Prefetching for unscheduled module CandPtrProjector/'pfCandsNoJetsNoEleNoMu'
   [145] Prefetching for unscheduled module CandPtrProjector/'pfCandsNoJetsNoEle'
   [146] Prefetching for unscheduled module CandPtrProjector/'pfCandsNoJets'
   [147] Prefetching for unscheduled module ShiftedParticleMETcorrInputProducer/'shiftedPatMETCorrJetEnDown'
   [148] Prefetching for unscheduled module ShiftedPATJetProducer/'shiftedPatJetEnDown'
   [149] Prefetching for unscheduled module ShiftedParticleMETcorrInputProducer/'shiftedPatMETCorrJetEnUp'
   [150] Prefetching for unscheduled module ShiftedPATJetProducer/'shiftedPatJetEnUp'
   [151] Prefetching for unscheduled module PATPFJetMETcorrInputProducer/'patPFMetT1T2Corr'
   [152] Prefetching for unscheduled module PATJetCleaner/'cleanedPatJets'
   [153] Prefetching for unscheduled module PATElectronSelector/'selectedPatElectrons'
   [154] Calling method for unscheduled module PATElectronProducer/'patElectrons'
Exception Message:
ValueMap: no associated value for given product and index

Any ideas?

ishvetso commented:
one more comment: the problem seems to be partly related to the logic for PATElectronProducer, the line that was added:
https://github.com/ishvetso/cmssw/blob/3c0caae6f02a2cab7011c808998ff42279e86fc8/PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc#L522

in fact, that's a different collection of electrons, seems like logic is not correct?

Now I get a different problem:
....
   [153] Calling method for unscheduled module PATElectronSelector/'selectedPatElectrons'
Exception Message:
A std::bad_alloc exception was thrown.
The job has probably exhausted the virtual memory available to the process.
----- End Fatal Exception -------------------------------------------------

monttj commented:
Hi Ivan,

I am wondering if it is related with the "egmElectronIsolationAODPUPPI" value map which does not have the surrounding particles, packedPFCandidates?

https://github.com/cms-sw/cmssw/pull/15528/files#diff-3b959e30c434f42e3679324062a05fc3R26

Regards,
Taejeong

ishvetso commented:
Hi Taejenog.

no, that should be set here:
https://github.com/ishvetso/cmssw/blob/3c0caae6f02a2cab7011c808998ff42279e86fc8/PhysicsTools/PatAlgos/python/slimming/egmIsolationsPUPPI_cfi.py#L14-L18

Ivan

monttj commented:
Hi Ivan,

But that was only for miniAOD.
https://github.com/ishvetso/cmssw/blob/3c0caae6f02a2cab7011c808998ff42279e86fc8/PhysicsTools/PatAlgos/python/slimming/egmIsolationsPUPPI_cfi.py#L3

In PAT object, you read "egmElectronIsolationAODPUPPI" as input?
I was not able to find the place where egmElectronIsolationAODPUPPI was set with the surrounding particles. 

Taejeong

ishvetso commented:
Hi Taejenog,

this is happening here:

https://github.com/ishvetso/cmssw/blob/3c0caae6f02a2cab7011c808998ff42279e86fc8/PhysicsTools/PatAlgos/python/producersLayer1/electronProducer_cfi.py#L40-L42

Regards,
Ivan.

ishvetso committed:
Merged PUPPI_isolation_electrons_miniAOD from repository ishvetso

ishvetso committed:
remove userFloat for puppi iso

cmsbuild commented:
Pull request #15528 was updated. @ghellwig, @vanbesien, @cvuosalo, @franzoni, @cerminar, @fabozzi, @cmsbuild, @rekovic, @srimanob, @montjj, @slava77, @ggovi, @mmusich, @dmitrijus, @hengne, @mulhearn, @davidlange6 can you please check and sign again.

ishvetso committed:
fix name

cmsbuild commented:
Pull request #15528 was updated. @ghellwig, @vanbesien, @cvuosalo, @franzoni, @cerminar, @fabozzi, @cmsbuild, @rekovic, @srimanob, @montjj, @slava77, @ggovi, @mmusich, @dmitrijus, @hengne, @mulhearn, @davidlange6 can you please check and sign again.

slava77 commented:
-1

needs a rebase (to get rid of all unrelated merges)

ishvetso commented:
can you tell me how this should be exactly done?


26.08.16 23:23, Slava Krutelyov пишет:
>
> -1
>
> needs a rebase (to get rid of all unrelated merges)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/15528#issuecomment-242855358>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFMRQmlJhIvcIISA_UO9XVd0A5-LcPZBks5qj1lNgaJpZM4JpENC>.
>

-- 
Best regards,
Ivan Shvetsov.

slava77 commented:
for rebase:
http://cms-sw.github.io/tutorial-forward-port.html
look for "rebase"

ishvetso committed:
temp? ...

cmsbuild commented:
Pull request #15528 was updated. @ghellwig, @vanbesien, @cvuosalo, @franzoni, @cerminar, @fabozzi, @cmsbuild, @rekovic, @srimanob, @montjj, @slava77, @ggovi, @mmusich, @dmitrijus, @hengne, @mulhearn, @davidlange6 can you please check and sign again.

ishvetso committed:
fix name

cmsbuild commented:
Pull request #15528 was updated. @ghellwig, @vanbesien, @cvuosalo, @franzoni, @cerminar, @fabozzi, @cmsbuild, @rekovic, @srimanob, @montjj, @slava77, @ggovi, @mmusich, @dmitrijus, @hengne, @mulhearn, @davidlange6 can you please check and sign again.

ishvetso commented:
I did rebase : 
git rebase CMSSW_8_1_X_2016-08-25-2300

but I don't think it worked ...

slava77 commented:
try
```
git rebase --onto CMSSW_8_1_X_2016-08-25-2300 CMSSW_8_1_X_2016-08-25-2300
```

slava77 commented:
... it worked for me
https://github.com/cms-sw/cmssw/compare/CMSSW_8_1_X...slava77:pull15528-8df0152-rebCMSSW_8_1_X_2016-08-26-1100

slava77 commented:
-1
to be rebased

ishvetso commented:
I'm following what you wrote here:

https://github.com/cms-sw/cmssw/issues/11426

 to be more precise:
1. start with new area, cmsrel CMSSW_8_1_X_2016-08-25-2300 etc
2. git cmd-init
3. git fetch official-cmssw CMSSW_8_1_X_2016-08-25-2300:CMSSW_8_1_X_2016-08-25-2300
4. git fetch my-cmssw PUPPI_isolation_electrons_miniAOD:PUPPI_isolation_electrons_miniAOD_rebased
5. git cms-sparse-checkout CMSSW_8_1_X_2016-08-19-1100 PUPPI_isolation_electrons_miniAOD_rebased
6. git checkout PUPPI_isolation_electrons_miniAOD_rebased
7. git rebase CMSSW_8_1_X_2016-08-25-2300

then I simply do 

git push my-cmssw PUPPI_isolation_electrons_miniAOD_rebased

I think that's fine? 
I got a warning:
[ishvetso@lxplus135 src]$ git rebase CMSSW_8_1_X_2016-08-25-2300
warning: refname 'CMSSW_8_1_X_2016-08-25-2300' is ambiguous.
warning: refname 'CMSSW_8_1_X_2016-08-25-2300' is ambiguous.

is it fine?

ishvetso commented:
rebased, please refer to #15628

slava77 commented:
please close
it is now superseded by #15628
